staging/src/k8s.io/apiserver/pkg/admission/plugin/policy/generic/policy_source.go

```diff
@@ -41,6 +41,10 @@ import (
 	"k8s.io/klog/v2"
 )
 
+var (
```

Comment:
add a comment indicating this is exported so it can be override on the tests

Reply:
Done. Though another option as a short term fix, as suggested by @cici37, is to introduce a time.Sleep() within the test, if we do not feel comfortable exporting this value just for the tests to pass. Unsure which way is the preferred one here. Open to suggestions @aojea @BenTheElder 

@jpbetz suggested to eventually have checks that can detect when the policy changes took effect which is an ideal solution to the unnecessary delay + flakiness that might arise from time.sleep() or policy-refresh operation. Included this in the comment as well.

Reply:
I think we should avoid exporting this because someone will depend on the API

maybe we export something that indirectly does this and is clearly marked as for tests.

It would be nice if go had support for manipulating time related methods for tests and we might want to consider that in utils like `wait` at some point.

will think on this a bit.

Reply:
one quick hack might be to make this var in an `internal/` package so only our tests can even access it?

Reply:
Hmm but looks like vars in `internal` package can only be imported by code rooted in parent directoy of the internal package? From [Go docs](https://go.dev/doc/go1.4#internalpackages) :
```
When the go command sees an import of a package with internal in its path, it verifies that the package doing the import is within the tree rooted at the parent of the internal directory. For example, a package .../a/b/c/internal/d/e/f can be imported only by code in the directory tree rooted at .../a/b/c. It cannot be imported by code in .../a/b/g or in any other repository.
```  

This doesnt solve our use case where we are trying to overwrite this value for our integration tests..

Reply:
It does, you just have to choose the internal root appropriately (e.g. it could be the root of the repo, but that would be a big change)

Reply:
raised to #k8s-code-organization for input

Reply:
> I think we should avoid exporting this because someone will depend on the API
> 
> maybe we export something that indirectly does this and is clearly marked as for tests.

+1 to a setter clearly documented / labeled as test-only. We do this in other places

```
git grep -E 'func Set.*Test'
cmd/kube-apiserver/app/server.go:func SetServiceResolverForTests(resolver webhook.ServiceResolver) func() {
pkg/capabilities/capabilities.go:func SetForTests(c Capabilities) {
staging/src/k8s.io/apiserver/pkg/server/config.go:func SetHostnameFuncForTests(name string) {
staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/config.go:func SetKDFForTests(b bool) func() {
```

etc

Reply:
(don't forget to find a way to reset to the normal default after the test is done)

Reply:
Thanks for the tip! Made changes as suggested. Could you PTAL?

Reply:
This part LGTM, thanks!